### PR TITLE
Var-arg pattern (#59)

### PR DESCRIPTION
**Pull request by gosubpl**
Refs #59

### CODE REVIEW

File: readme/Rewrites.scalatex [hunk 1]
```diff
@@ -139,6 +139,36 @@
       Such members already have a return type of @code{Unit} and sometimes this is unexpected.
       Adding an explicit return type makes it more obvious.
 
+  @sect(DottyVarArgPattern.toString)
+    @p
+      Replaces @code{@} symbols in VarArg patterns with a colon (@code{:}). See @lnk{http://dotty.epfl.ch/docs/reference/changed/vararg-patterns.html}
```
Comment (olafurpg) [inline readme/Rewrites.scalatex]: This line here has a syntax error, I think the `@` symbol needs to be escaped with `@@`. To test locally run `readme/run`.

File: readme/Rewrites.scalatex [hunk 2]
```diff
@@ -139,6 +139,36 @@
...
+
+    @hl.scala
+      // before
+      package test
+
+      object VarArg {
+        List(1, 2, 3, 4) match {
+          case List(1, 2, xs @ _*) =>
```
Comment (olafurpg) [inline readme/Rewrites.scalatex]: Can we minimize this example to 

```
// before
case List(1, 2, xs @ _*)
// after
case List(1, 2, _*)
```

File: scalafix-tests/input/src/main/scala/test/VarArg.scala [hunk 3]
```diff
@@ -0,0 +1,14 @@
+/*
+rewrites = VarArg
+ */
+package test
+
+object VarArg {
+  List(1, 2, 3, 4) match {
+    case List(1, 2, xs @ _*) =>
+      val ys: Seq[Int] = xs
```
Comment (olafurpg) [inline scalafix-tests/input/src/main/scala/test/VarArg.scala]: Remove this line and below? We should try to have as small tests as possible.

File: scalafix-core/src/main/scala/scalafix/rewrite/VarArg.scala [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package scalafix.rewrite
+
+import scala.meta._
+
+import scala.meta.tokens.Token
+import scalafix.Patch
+
+case object VarArg extends Rewrite {
```
Comment (olafurpg) [inline scalafix-core/src/main/scala/scalafix/rewrite/VarArg.scala]: Rename to `DottyVarArgPattern`.

**Comment by gosubpl** [comment]
Ready for the second pass :)

File: readme/Rewrites.scalatex [hunk 5]
```diff
@@ -139,6 +139,16 @@
...
 
+  @sect(DottyVarArgPattern.toString)
+    @p
+      Replaces @code{@@} symbols in VarArg patterns with a colon (@code{:}). See @lnk{http://dotty.epfl.ch/docs/reference/changed/vararg-patterns.html}
+
+    @hl.scala
+      // before
+      case List(1, 2, xs @@ _*)
+      // after
+      case List(1, 2, _*)
```
Comment (olafurpg) [inline readme/Rewrites.scalatex]: Missing colon?

**Comment by olafurpg** [comment]
I tracked down the issue to a configuration error in the ci build script. I opened #189 to fix this.

**Comment by olafurpg** [comment]
Also, please run `./scalafmt` to fix the build failures.

**Comment by olafurpg** [comment]
Please rebase on top of master and see if that fixes the CI errors.

**Comment by gosubpl** [comment]
Many thanks @olafurpg ! Rebased & the doc error fixed (plus a minor sbt build issue for local builds). We'll see if it builds :)

**Comment by olafurpg** [comment]
Cool! Looks like all the tests pass. Only thing missing is to run./scalafmt and we are good to go!

**Comment by gosubpl** [comment]
Sorry 'bout the fmt, fixed.

**Comment by gosubpl** [comment]
@olafurpg many thanks for all your help, and thanks for your patience, this was my first PR for `scalafix`. I hope to add more soon!

**Pull request merged**